lib/internal/fs/promises.js

```diff
@@ -632,10 +635,13 @@ async function writeFile(path, data, options) {
   }
 
   if (path instanceof FileHandle)
-    return writeFileHandle(path, data);
+    return writeFileHandle(path, data, options.signal);
 
   const fd = await open(path, flag, options.mode);
-  return writeFileHandle(fd, data).finally(fd.close);
+  if (options.signal && options.signal.aborted) {
+    throw new lazyDOMException('The operation was aborted', 'AbortError');
+  }
+  return writeFileHandle(fd, data, options.signal).finally(fd.close);
```

Comment:
```suggestion
  return PromisePrototypeFinally(writeFileHandle(fd, data, options.signal), fd.close);
```

Reply:
Wait, actually that's an unrelated change to this PR - mind if I use this for a code-and-learn type thing with a first time contributor?

Reply:
Sure, please do!

Reply:
Thanks, working on it #35995